lib/Service/Provisioning/Manager.php

```diff
@@ -319,19 +319,26 @@ public function deprovision(Provisioning $provisioning): void {
 	/**
 	 * @param Provisioning[] $provisionings
 	 */
-	public function updatePassword(IUser $user, string $password, array $provisionings): void {
+	public function updatePassword(IUser $user, ?string $password, array $provisionings): void {
 		try {
 			$account = $this->mailAccountMapper->findProvisionedAccount($user);
 
 			$provisioning = $this->findMatchingConfig($provisionings, $user);
 			if ($provisioning === null) {
 				return;
 			}
-			$masterPassword = $provisioning->getMasterPassword();
-			$masterPasswordEnabled = $provisioning->getMasterPasswordEnabled();
-			if ($masterPasswordEnabled && $masterPassword !== null) {
-				$password = $masterPassword;
+
+			// FIXME: Need to check for an empty string here too?
+			// The password is empty (and not null) when using WebAuthn passwordless login.
+			// Maybe research other providers as well.
+			// Ref \OCA\Mail\Controller\PageController::index()
+			//     -> inital state for password-is-unavailable
+			if ($provisioning->getMasterPasswordEnabled() === true && $provisioning->getMasterPassword() !== null) {
+				$password = $provisioning->getMasterPassword();
 				$this->logger->debug('Password set to master password for ' . $user->getUID());
+			} else if ($password === null) {
```

Comment:
```suggestion
			} elseif ($password === null) {
```

one last cosmetic change. my bad